app/src/composables/use-local-storage-mutex.ts

```diff
@@ -1,51 +1,48 @@
-import { sleep } from '@directus/sdk';
+import { sleep } from "@directus/sdk";
 
 const MutexKey = ['auth_refresh'] as const;
 
 type MutexKey = (typeof MutexKey)[number];
 
+const timeout = 50;
+const maxRetries = 10;
+
 export function useLocalStorageMutex(key: MutexKey, expiresMs: number) {
 	const internalKey = `directus-mutex-${key}`;
-	const useWebLock = window.isSecureContext;
+	const useWebLock = !!navigator.locks;
 
-	async function acquireMutex() {
-		let releaseMutex: (() => void) | undefined;
+	let retries = 0;
 
+	async function acquireMutex(callback: (lock?: Lock | null) => Promise<any>): Promise<any> {
 		if (useWebLock) {
-			await new Promise<void>((resolve) => {
-				const acquireWebLock = () => {
-					navigator.locks.request(internalKey, { ifAvailable: true }, async function (granted) {
-						await new Promise<void>((lockResolve) => {
-							if (granted) {
-								releaseMutex = lockResolve;
-								resolve();
-							} else {
-								setTimeout(acquireWebLock, 500);
-							}
-						});
-					});
-				};
-
-				acquireWebLock();
-			});
-		} else {
-			// Random wait to prevent concurrent refreshes across browser windows/tabs
-			await sleep(Math.random() * 1000);
```

Comment:
@br41nslug This random delay was added for localstorage mutex to resolve the issue where multiple tabs reload concurrently and both could "acquire" a lock, resulting in the same race condition. Unfortunately, this random delay seems necessary unless there's another way to resolve it. 🤔 

With the random delay removed, the user would either be logged out or face the invalid token error.

![image](https://github.com/directus/directus/assets/26413686/bbeab5c0-b642-4c4c-bddd-9ed799eae353)